contracts/CollectiveGovernance.sol

```diff
@@ -0,0 +1,106 @@
+
+ * 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.
+
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+pragma solidity ^0.8.15;
+
+import "./GovernanceStorage.sol";
+import "./Governance.sol";
+import "./UpgradeableGovernance.sol";
+import "./VoteStrategy.sol";
+import "./ElectorVoterPoolStrategy.sol";
+
+/// @title CollectiveGovernance
+// factory contract for governance
+contract CollectiveGovernance is UpgradeableGovernance, Governance {
+    address private owner;
+
+    Storage private _storage;
+    VoteStrategy private _voteStrategy;
+
+    constructor() {
+        owner = msg.sender;
+        _storage = new GovernanceStorage();
+        emit StorageAddress(address(_storage));
+        _voteStrategy = new ElectorVoterPoolStrategy(_storage);
+        emit StrategyChange(0, _voteStrategy.version(), address(_voteStrategy));
+    }
+
+    modifier requireContractOwner() {
+        require(owner == msg.sender, "Not contract owner");
+        _;
+    }
+
+    function setVoteStrategy(address _strategy) external requireContractOwner {
+        uint32 version = _voteStrategy.version();
+        _voteStrategy = VoteStrategy(_strategy);
+        uint32 newVersion = _voteStrategy.version();
+        emit StrategyChange(version, newVersion, _strategy);
+    }
+
+    function getCurrentStrategyVersion() external view returns (uint32) {
+        return _voteStrategy.version();
+    }
+
+    function getCurrentStrategyAddress() external view returns (address) {
+        return address(_voteStrategy);
+    }
+
+    function getStorageAddress() external view returns (address) {
+        return address(_storage);
+    }
+
+    function propose(
+        uint256 quorumThreshold,
```

Comment:
proposer shouldn't be able to set their own quorum

Reply:
who would set the quorum, does it require a minimum quorum to prevent cheating or a 3rd party?

Reply:
We discussed adding a minimum settable by the parent contract